File: LCObjective-C/LCObjective-C/Problems/FirstOccurrenceString/FirstOccurrenceString.m
//
//  FirstOccurrenceString.m
//  LCObjective-C
//
//  Created by Sherzod Akhmedov on 6/29/24.
//

#import <Foundation/Foundation.h>
#import "FirstOccurrenceString.h"

@implementation FirstOccurrenceString

+(NSInteger) findPositionOfNeedle:(NSString *)needle fromHaystack: (NSString *)haystack {
    NSRange range = [haystack rangeOfString:needle];
    if (range.length == 0) {
        return -1;
    }
    return range.location;
}

@end

File: LCSwift/LCSwift/Problems/FirstOccurrenceString/FirstOccurrenceString.swift
//
//  FirstOccurrenceString.swift
//  LCSwift
//
//  Created by Sherzod Akhmedov on 6/29/24.
//

import Foundation

final class FirstOccurrenceString {
    static func strStr(_ haystack: String, _ needle: String) -> Int {
        guard let range = haystack.range(of: needle) else {
            return -1
        }
        return haystack.distance(from: haystack.startIndex, to: range.lowerBound)
    }
}
